### PR TITLE
Issue/v4/894 config db

### PR DESCRIPTION
**Pull request by stefanorosanelli**

### CODE REVIEW

File: plugins/BEdita/Core/src/Configure/Engine/DatabaseConfig.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        $query = $config->find()->select(['name', 'context', 'content']);
+        $query->where(function ($exp, $q) {
+            return $exp->notIn('name', $this->reservedKeys);
+        });
+        if ($key) {
+            $query->andWhere(['context' => $key]);
+        }
+        $results = $query->all();
+        foreach ($results as $data) {
+            //pr('config read: ' . print_r($data, true));
```
Comment (batopa) [inline plugins/BEdita/Core/src/Configure/Engine/DatabaseConfig.php]: comment to remove

File: plugins/BEdita/Core/src/Configure/Engine/DatabaseConfig.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+     * @param string $key The group parameter to write to.
+     * @param array $data The data to write.
+     * @return bool Success.
+     */
+    public function dump($key, array $data)
+    {
+        $context = $key;
+        $entities = [];
+        $table = TableRegistry::get('Config');
+        foreach ($data as $name => $content) {
```
Comment (batopa) [inline plugins/BEdita/Core/src/Configure/Engine/DatabaseConfig.php:80]: Maybe could be semplified as

``` php
foreach ($data as $name => $content) {
    if (in_array($name, $this->reservedKeys)) {
        continue;
    }

    $content = is_array($content) ? json_encode($content) : $this->valueToString($content);
    $entities[] = $table->newEntity(compact('name', 'context', 'content'));
}
```

Comment (stefanorosanelli) [inline plugins/BEdita/Core/src/Configure/Engine/DatabaseConfig.php:80]: :+1: 
change committed

**Comment by batopa** [comment]
Ok, remove the comment `//pr('config read: ' . print_r($data, true));` at line 58, then I can merge it

File: plugins/BEdita/Core/src/Configure/Engine/DatabaseConfig.php (hunk 1 above)
Comment (stefanorosanelli) [inline plugins/BEdita/Core/src/Configure/Engine/DatabaseConfig.php]: done

**Pull request merged**